control/plugin_manager.go

```diff
@@ -180,51 +180,51 @@ type loadedPlugin struct {
 
 // Name returns plugin name
 // implements the CatalogedPlugin interface
-func (lp *loadedPlugin) Name() string {
+func (lp loadedPlugin) Name() string {
 	return lp.Meta.Name
 }
 
 // PluginPath returns the plugin path
-func (lp *loadedPlugin) PluginPath() string {
+func (lp loadedPlugin) PluginPath() string {
 	return lp.Details.Path
 }
 
 // Key returns plugin type, name and version
-func (lp *loadedPlugin) Key() string {
+func (lp loadedPlugin) Key() string {
 	return fmt.Sprintf("%s"+core.Separator+"%s"+core.Separator+"%d", lp.TypeName(), lp.Name(), lp.Version())
 }
 
 // Version returns plugin version
 // implements the CatalogedPlugin interface
-func (lp *loadedPlugin) Version() int {
+func (lp loadedPlugin) Version() int {
 	return lp.Meta.Version
 }
 
 // TypeName returns plugin type as a string
 // implements the CatalogedPlugin interface
-func (lp *loadedPlugin) TypeName() string {
+func (lp loadedPlugin) TypeName() string {
 	return lp.Type.String()
 }
 
 // Status returns current plugin state
 // implements the CatalogedPlugin interface
-func (lp *loadedPlugin) Status() string {
+func (lp loadedPlugin) Status() string {
 	return string(lp.State)
 }
 
 // IsSigned returns plugin signing as a bool
 // implements the CatalogedPlugin interface
-func (lp *loadedPlugin) IsSigned() bool {
+func (lp loadedPlugin) IsSigned() bool {
 	return lp.Details.Signed
 }
 
 // LoadedTimestamp returns a unix timestamp of the LoadTime of a plugin
 // implements the CatalogedPlugin interface
-func (lp *loadedPlugin) LoadedTimestamp() *time.Time {
+func (lp loadedPlugin) LoadedTimestamp() *time.Time {
 	return &lp.LoadedTime
 }
 
-func (lp *loadedPlugin) Policy() *cpolicy.ConfigPolicy {
+func (lp loadedPlugin) Policy() *cpolicy.ConfigPolicy {
```

Comment:
@IRCody I think it is similar to below example
```go
type Fooer interface {
	foo()
}

type Foo struct {}
func (f *Foo) foo(){}

func doStuff(){
  var fi Fooer
  // works ok since *Foo implements interface Fooer
  fi = &Foo{}
  fi.foo()
  
  // Compile error, Foo does not implement Fooer (foo method has pointer receiver)
  fi = Foo{}
  fi.foo()
}
```
If there is something I'm missing, please let me know.